internal/providers/terraform/aws/nat_gateway.go

```diff
@@ -15,11 +15,15 @@ func GetNATGatewayRegistryItem() *schema.RegistryItem {
 func NewNATGateway(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
 	region := d.Get("region").String()
 
-	args := &aws.NATGatewayArguments{
-		Address: d.Address,
-		Region:  region,
-	}
-	args.PopulateUsage(u)
+	// We initialize the args and populate the defaults and usage data.
+	args := &aws.NATGatewayArguments{}
+	args.PopulateArgs(u)
+
+	// Then we can override the fields with IAC fields.
```

Comment:
I was thinking about this the other way around: the usage file could override the values pulled from IAC.   Maybe it would just be enough to do what you suggest and log that "Region from the usage file was ignored".

Reply:
That makes sense, I'm not sure whichever works better. I mean what source of information is more reliable for us, IAC or usage-file.

Reply:
I think there's cases for both.
1. The user might want to specify a default region in the usage file because we can't always get that from the IaC code.
2. The user might want to use the usage file to override the IaC code to get estimates for different instance counts in an autoscaling group.

For the no 1 use case, maybe `default_region` could just be a special case that we have a separate usage file key for.

Reply:
@sinabakh what did we decide for this? I think region should be treated the same way as every other field in the usage file.

Reply:
My vote is that the usage file should take precedence over IaC.  In other words, an arg calculated from IaC should only be used if the usage file value is empty.

I think the trick will be in making sure the sync_usage stuff only writes defaults for args that don't normally come from IaC.

Reply:
@aliscott Sorry I missed your comment. I agree with @tim775